apps/client/src/app/pages/map/hooks/useDistanceByBike.tsx

```diff
@@ -0,0 +1,28 @@
+import { LngLat } from 'mapbox-gl';
+
+interface DirectionsResponse {
+  routes: {
+    distance: number;
+  }[];
+}
+
+const calculateDistanceByBike = async (
+  start: LngLat,
+  end: LngLat
+): Promise<number> => {
+  const response = await fetch(
+    `https://api.mapbox.com/directions/v5/mapbox/cycling/${start.lat},${
+      start.lng
+    };${end.lat},${end.lng}?access_token=${
+      import.meta.env.VITE_PUBLIC_MAPBOX_TOKEN
+    }`
+  );
+
+  const data = (await response.json()) as DirectionsResponse;
+
+  const { distance } = data.routes[0];
+  const distanceInKilometers = Math.round((distance / 1000) * 100) / 100;
```

Comment:
Add equations to a variable with a descriptive name to make it possible to know what the equations do and make it easier to read.